### DISCUSSION TITLE
Is there any way to send system message to Claude (along with user message)?

### DISCUSSION THREAD

**Discussion by lofti198** [higty/higlabo#110] [Q&A] [answered]
Is there any way to send system instructions to Claude (along with user message)? In their ChatMessageRole there are just User and Assistant available. Or it is common practice to send instructions together with main prompt in User message?

![image](https://github.com/higty/higlabo/assets/42960488/3d0cfe78-3526-45bb-9fde-2c01d916b6ea)

**Comment by higty** [comment, marked as answer]
See this document.
https://docs.anthropic.com/en/api/messages

Maybe System property is that.
Please try it.

**Reply by lofti198** [reply]
Thank you for the hint! I found an example here https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/system-prompts:
```
response = client.messages.create(
    model="claude-3-5-sonnet-20240620",
    max_tokens=2048,
    system="You are a seasoned data scientist at a Fortune 500 company.", # <-- role prompt
    messages=[
        {"role": "user", "content": "Analyze this dataset for anomalies: <dataset>{{DATASET}}</dataset>"}
    ]
)
```

And thank you for using this parameter in API:
```
var cl = new AnthropicClient(Environment.GetEnvironmentVariable("CLAUDE_API_KEY"));

var p = new HigLabo.Anthropic.MessagesParameter();

p.Messages.Add(new HigLabo.Anthropic.ChatMessage(HigLabo.Anthropic.ChatMessageRole.User, text));
p.Model = ModelNames.Claude3Opus;
p.Max_Tokens = 1000;
**p.System = instruction;**
var res = await cl.MessagesAsync(p);
```

**Reply by higty** [reply]
That is great! Happy programing!
